src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -36,6 +39,7 @@ public Socket(EventBus eventBus, SocketHint<T> socketHint, T value) {
         this.eventBus = eventBus;
         this.socketHint = socketHint;
         this.value = value;
+        this.direction = Direction.INPUT;
```

Comment:
Operations construct sockets. You know when you are constructing a socket whether or not it is going to be an input or an output.